hax/hax/motr/__init__.py

```diff
@@ -108,6 +110,36 @@ def _entrypoint_request_cb(self, reply_context: Any, req_id: Any,
                   " '{}', process fid = {}".format(remote_rpc_endpoint,
                                                    str(process_fid)) +
                   ' The request will be processed in another thread.')
+        try:
+            cns = ConsulUtil()
+            if is_first_request and (not cns.is_proc_client(process_fid)):
+                # This is the first start of this process or the process has
+                # restarted.
+                # Let everyone know that the process has restarted so that
+                # they can re-establish their connections with the process.
+                # Disconnect all the halinks this process is any.
+                # Motr clients are filtered out as they are the initiators of
+                # the rpc connections to motr ioservices. Presently there's no
+                # need identified to report motr client restarts, Consul will
+                # anyway detect the failure and report the same so we exclude
+                # reporting the same during their first entrypoint request.
+                # But we need to do it for motr server processes.
+                q: Queue = Queue(1)
+                LOG.debug('first entrypoint request, broadcasting FAILED')
+                self.queue.put(
+                    BroadcastHAStates(
+                        states=[HAState(fid=process_fid,
+                                        status=ServiceHealth.FAILED)],
+                        reply_to=q))
+                ids: List[MessageId] = q.get()
+                LOG.debug('waiting for broadcast of %s ep: %s',
+                          ids, remote_rpc_endpoint)
+                self.herald.wait_for_all(HaLinkMessagePromise(ids))
+        except Exception:
+            pass
```

Comment:
Shouldn't we log at least something here?

Reply:
@azheregelya if you mean a log message about waiting, it is done in the `wait_for_all()` function.

Reply:
I mean entire try...except block here is to silence any Exception-class-derived exception.
Shouldn't we at least log this in case something is not handled?